pkg/partmgr/partmgr.go

```diff
@@ -18,123 +18,203 @@ under the Apache 2.0 license is conditioned upon your compliance with
+			} else {
+				return p.headPartition, nil
+			}
+		} else {
+			//iterate backwards, ignore last elem as it's the headPartition
+			for i := len(p.partitions) - 2; i >= 0; i-- {
+				if t > p.partitions[i].startTime {
+					return p.partitions[i], nil
+				}
+			}
+		}
+	}
+	return p.headPartition, nil
 }
 
-func (p *PartitionManager) PartsForRange(mint, maxt int64) []*DBPartition {
-
-	return []*DBPartition{p.headPartition}
+func (p *PartitionManager) createNewPartition(t int64) (*DBPartition, error) {
+	time := t & 0x7FFFFFFFFFFFFFF0
+	partPath := path.Join(p.path, strconv.FormatInt(time/1000, 10)) + "/"
+	partition := NewDBPartition(p, time, partPath)
+	p.currentPartitionInterval = partition.partitionInterval
+	p.headPartition = partition
+	p.partitions = append(p.partitions, partition)
+	err := p.updatePartitionInSchema(partition)
+	return partition, err
 }
 
-func (p *PartitionManager) GetHead() *DBPartition {
+func (p *PartitionManager) updatePartitionInSchema(partition *DBPartition) error {
+	p.cfg.Partitions = append(p.cfg.Partitions, config.Partition{StartTime: partition.startTime, SchemaInfo: p.cfg.PartitionSchemaInfo})
+	data, err := json.Marshal(p.cfg)
+	if err != nil {
+		return errors.Wrap(err, "Failed to update new partition in schema file")
+	}
+	err = p.container.Sync.PutObject(&v3io.PutObjectInput{Path: path.Join(p.path, config.SCHEMA_CONFIG), Body: data})
+	return err
+}
 
-	return p.headPartition
+func (p *PartitionManager) PartsForRange(mint, maxt int64) []*DBPartition {
+	var parts []*DBPartition
+	for _, part := range p.partitions {
+		if part.startTime+p.currentPartitionInterval >= mint && (maxt == 0 || part.startTime <= maxt) {
+			parts = append(parts, part)
+		}
+	}
+	return parts
 }
 
 type DBPartition struct {
-	manager        *PartitionManager
-	path           string             // Full path (in the DB) to the partition
-	partID         int                // PartitionID
-	startTime      int64              // Start from time/date
-	days           int                // Number of days stored in the partition
-	hoursInChunk   int                // number of hours stored in each chunk
-	prefix         string             // Path prefix
-	retentionDays  int                // Keep samples for N days
-	defaultRollups aggregate.AggrType // Default Aggregation functions to apply on sample update
-	rollupTime     int64              // Time range per aggregation bucket
-	rollupBuckets  int                // Total number of buckets per partition
+	manager           *PartitionManager
+	path              string             // Full path (in the DB) to the partition
+	startTime         int64              // Start from time/date
+	partitionInterval int64              // Number of millis stored in the partition
+	chunkInterval     int64              // number of millis stored in each chunk
+	prefix            string             // Path prefix
+	retentionDays     int                // Keep samples for N hours
```

Comment:
fix comment, why duration units are changing all over the way?
Note, string/int/ int64... if we decided to define duration in milliseconds, lets do it all the way. String should be used in configuration and CLI (human interfaces). internal representation should be in `millis` only

Reply:
As said in other comment, will fix in a different PR